src/schema/users.ts

```diff
@@ -773,31 +800,8 @@ export const getMarketingCta = async (
   }
 
   // If the vale in redis is `EMPTY`, we fallback to `null`
-  let marketingCta: MarketingCta | null = JSON.parse(rawRedisValue || null);
-
-  // If the key is not in redis, we need to fetch it from the database
-  if (!marketingCta) {
-    const userMarketingCta = await con.getRepository(UserMarketingCta).findOne({
-      where: {
-        userId,
-        readAt: IsNull(),
-        marketingCta: {
-          status: MarketingCtaStatus.Active,
-        },
-      },
-      order: { createdAt: 'ASC' },
-      relations: ['marketingCta'],
-    });
-
-    marketingCta = userMarketingCta?.marketingCta || null;
-    const redisValue = userMarketingCta
-      ? JSON.stringify(marketingCta)
-      : RedisMagicValues.SLEEPING;
-
-    await setRedisObjectWithExpiry(redisKey, redisValue, ONE_WEEK_IN_SECONDS);
-  }
-
-  return marketingCta;
+  const marketingCta: MarketingCta | null = JSON.parse(rawRedisValue || null);
+  return marketingCta || cachePrefillMarketingCta(con, userId, redisKey);
```

Comment:
Didn't need this change, should I keep it or revert it?

Reply:
If we don't need it lets remove. Less code is better.

Reply:
Reverted it